app/gui/src/presenter/graph.rs

```diff
@@ -619,16 +653,22 @@ impl Graph {
 
             reset_node_types <- any(update_node_expression, init_node_expression)._0();
             set_expression_type <= reset_node_types.map(f!((view_id) model.all_types_of_node(*view_id)));
-            set_method_pointer <= reset_node_types.map(f!((view_id) model.all_method_pointers_of_node(*view_id)));
             view.set_expression_usage_type <+ set_expression_type;
-            view.set_method_pointer <+ set_method_pointer;
 
             update_expressions <- source::<Vec<ast::Id>>();
             update_expression <= update_expressions;
             view.set_expression_usage_type <+ update_expression.filter_map(f!((id) model.refresh_expression_type(*id)));
-            view.set_method_pointer <+ update_expression.filter_map(f!((id) model.refresh_expression_method_pointer(*id)));
             view.set_node_error_status <+ update_expression.filter_map(f!((id) model.refresh_node_error(*id)));
 
+            // Refreshing expression widgets
+            widgets_to_update <- any(...);
+            widgets_to_update <+ reset_node_types.map(f!((view_id) model.refresh_all_widgets_of_node(*view_id))).iter();
+            widgets_to_update <+ update_expression.filter_map(f!((id) model.refresh_expression_widgets(*id)));
+            widgets_to_update <+ view.widgets_requested;
+            widget_request <- widgets_to_update.filter_map(f!(((node, call, target)) model.update_widget_request_data(*node, *call, *target)));
+            widgets.request_widgets <+ widget_request;
+            view.set_expression_widgets <+ widgets.widget_data;
+            widgets.remove_node <+ view.node_removed;
```

Comment:
move it to a function. I know the rest of the code is ugly and not refactored, but lets keep new code tidy :)